# AI Tools Declaration

In [ ]:
import numpy as np

X_train = np.loadtxt("Xtrain.csv")
Y_train = np.loadtxt("Ytrain.csv")
X_test = np.loadtxt("Xtest.csv")

# Prediction Metrics 

# Question 1

## Dataset Overview

In [6]:
np.unique_counts(Y_train)

UniqueCountsResult(values=array([-1.,  1.]), counts=array([2386,  614]))

a) Training set: There are 2386 negative examples, 614 positive examples. Test set: Unknown.

b) There is some class imbalance in the training data, with a N:P ratio of about 4:1. This causes some potential issues in designing a machine learning solution. With an overrepresented negative class, a trained model may exhibit high specificity but low precision, i.e. it focuses more on correctly predicting the negative class. There may be a high number of false negative predictions. 